### ISSUE TITLE
hyphens in date ranges broke in 1.19

### ISSUE THREAD

**Issue by simonmichael** [simonmichael/hledger#1363]
Since 1.19 hledger displays date ranges as DATE..DATE instead of DATE-DATE. However the old syntax is supposed to still be supported on the command line.
```
$ hledger-1.19 print date:2017-2018
hledger-1.19: "date:2017-2018" gave a date parse error ()
```

**Comment by Xitian9** [comment]
Interestingly, the correct result is obtained by each of the following commands:
```hledger print date:"2017 - 2018"```
```hledger print date:"2017- 2018"```
```hledger print date:"2017 -2018"```
The parse error is only given when there is no space padding on either side of the hyphen. I'll try to figure out the source of this error.

**Comment by simonmichael** [comment]
Fixed by #1366.
